plugins/repository-hdfs/build.gradle

```diff
@@ -0,0 +1,175 @@
+      exclude group: "com.sun.jersey.jersey-test-framework"
+      exclude module: "guice"
+      exclude group: "org.mortbay.jetty"
+      exclude group: "tomcat"
+      exclude module: "commons-el"
+      exclude module: "hsqldb"
+      exclude group: "org.eclipse.jdt"
+      exclude module: "commons-beanutils"
+      exclude module: "commons-beanutils-core"
+      exclude module: "javax.servlet"
+      exclude module: "guava"
+      exclude module: "junit"
+      // provided by ES itself
+      exclude group: "org.slf4j"
+      exclude group: "log4j"
+  }
+
+  hadoop2("org.apache.hadoop:hadoop-hdfs:${versions.hadoop2}") {
+      exclude module: "guava"
+      exclude module: "log4j"
+      exclude module: "junit"
+      // provided by ES itself
+      exclude group: "org.slf4j"
+      exclude group: "log4j"
+  }
+} 
+
+configurations.all {
+    resolutionStrategy {
+        force "commons-codec:commons-codec:${versions.commonscodec}"
+        force "commons-logging:commons-logging:${versions.commonslogging}"
+        force "commons-lang:commons-lang:2.6"
+        //force "commons-collections:commons-collections:3.2.2"
+        force "commons-httpclient:commons-httpclient:3.0.1"
+        force "org.codehaus.jackson:jackson-core-asl:1.8.8"
+        force "org.codehaus.jackson:jackson-mapper-asl:1.8.8"
+    }
+}
+
+
+dependencyLicenses {
+  mapping from: /hadoop-.*/, to: 'hadoop'
+}
+
+compileJava.options.compilerArgs << '-Xlint:-deprecation,-rawtypes'
+
+// main jar includes just the plugin classes
+jar {
+    include "org/elasticsearch/plugin/hadoop/hdfs/*"
+}
+
+// haddop jar (which actually depend on Hadoop)
+task hadoopLinkedJar(type: Jar, dependsOn:jar) {
+    appendix "internal"
+    from sourceSets.main.output.classesDir
+    // exclude plugin
+    exclude "org/elasticsearch/plugin/hadoop/hdfs/*"
+}
+
+
+// configure 'bundle' as being w/o Hadoop deps
+bundlePlugin {
+    into ("internal-libs") {
+        from hadoopLinkedJar.archivePath
+    }
+}
+
+task distZipHadoop1(type: Zip, dependsOn: [hadoopLinkedJar, jar]) { zipTask ->
+    from zipTree(bundlePlugin.archivePath)
+    description = "Builds archive (with Hadoop1 dependencies) suitable for download page."
+    classifier = "hadoop1"
+
+    into ("hadoop-libs") {
+        from configurations.hadoop1.allArtifacts.files
+        from configurations.hadoop1
+    }
+}
+
+task distZipHadoop2(type: Zip, dependsOn: [hadoopLinkedJar, jar]) { zipTask ->
+    from zipTree(bundlePlugin.archivePath)
+    description = "Builds archive (with Hadoop2/YARN dependencies) suitable for download page."
+    classifier = "hadoop2"
+
+    into ("hadoop-libs") {
+        from configurations.hadoop2.allArtifacts.files
+        from configurations.hadoop2
+    }
+}
+
+artifacts {
+    archives bundlePlugin
+    'default' bundlePlugin
+    archives distZipHadoop1
+    archives distZipHadoop2
+}
+
+integTest {
+    cluster {
+        plugin(pluginProperties.extension.name, zipTree(distZipHadoop1.archivePath))
```

Comment:
From what I can tell, this configuration doesn't change the plugin source. The bundlePlugin archive is still used instead, meaning the plugin without the Hadoop jars instead of the one with Hadoop1 in it. Which means the plugin loads but the comment somebody pokes at it, it will fail with ClassNotFoundException (since there are no Hadoop classes available).